### PR TITLE
Dev test

### PR DESCRIPTION
**Pull request by msimerson**
- test: switch runner from mocha to node builtin
- test: remove useless done
  - they are only required for callback style async tests
- pkg: rm .travis.yml
- ci: update test runners
- ci: automatically test node LTS versions
- ci: update lint workflow

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,15 +8,13 @@
...
   "dependencies": {},
   "devDependencies": {
-    "eslint": "*",
-    "mocha": "*",
     "uglify-es": "*"
   },
   "scripts": {
-    "lint": "npx eslint lib test",
-    "lintfix": "npx eslint --fix lib test",
+    "lint": "npx eslint@^8 lib test",
```
Comment (whitequark) [inline package.json:14]: I think I'd actually prefer having eslint pinned to `^8` in `devDependencies`?

File: package.json [hunk 2]
```diff
@@ -8,15 +8,13 @@
...
+    "lintfix": "npx eslint@^8 --fix lib test",
     "build": "npx uglifyjs --compress --mangle --wrap=window -o ipaddr.min.js lib/ipaddr.js",
-    "test": "npx _mocha"
+    "test": "npx mocha@^10"
```
Comment (whitequark) [inline package.json:17]: Happy to require node>=18 for development and ditch mocha, though.

**Review by whitequark — CHANGES_REQUESTED** [review]
Thanks for picking this up!

**Comment by msimerson** [comment]
Okay, so several changes here:

- dep(eslint): declared dependency on v8
- move the test file into lib (next to the code it runs). Naming files with `*.test.js` and putting them next to the files they test is apparently a pattern. Your thoughts?
- mocha -> `node --test`
  - ci: drop testing for older nodes that lack support for `node --test`

**Comment by msimerson** [comment]
> naming files with *.test.js and putting them next to the files they test is apparently a pattern.

Something I just thought of and don't like, is that putting the test file in lib publishes it on npm. That's a small deal, but with 36m weekly downloads, the smallness has a very large multiplier. I've talked myself out of it. 😲 Sorry for the noise.

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**